src/uu/ls/src/dired.rs

```diff
@@ -0,0 +1,178 @@
+// This file is part of the uutils coreutils package.
+//
+// For the full copyright and license information, please view the LICENSE
+// file that was distributed with this source code.
+// spell-checker:ignore dired subdired dired
+
+use crate::Config;
+use std::fmt;
+use std::io::{BufWriter, Stdout, Write};
+use uucore::error::UResult;
+
+#[derive(Debug, Clone)]
+pub struct BytePosition {
+    pub start: usize,
+    pub end: usize,
+}
+
+/// Represents the output structure for DIRED, containing positions for both DIRED and SUBDIRED.
+#[derive(Debug, Clone, Default)]
+pub struct DiredOutput {
+    pub dired_positions: Vec<BytePosition>,
+    pub subdired_positions: Vec<BytePosition>,
+    pub just_printed_total: bool,
+}
+
+impl fmt::Display for BytePosition {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        write!(f, "{} {}", self.start, self.end)
+    }
+}
+
+// When --dired is used, all lines starts with 2 spaces
+static DIRED_TRAILING_OFFSET: usize = 2;
+
+/// Calculates the byte positions for DIRED
+pub fn calculate_dired_byte_positions(
+    output_display_len: usize,
+    dfn_len: usize,
```

Comment:
What does `dfn` mean?

Reply:
Display File Name

we should rename it (it is from ls.rs)
I will create an issue